airsonic-main/src/main/java/org/airsonic/player/service/search/DocumentFactory.java

```diff
@@ -0,0 +1,175 @@
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Airsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2016 (C) Airsonic Authors
+ Based upon Subsonic, Copyright 2009 (C) Sindre Mehus
+ */
+
+package org.airsonic.player.service.search;
+
+import org.airsonic.player.domain.Album;
+import org.airsonic.player.domain.Artist;
+import org.airsonic.player.domain.MediaFile;
+import org.airsonic.player.domain.MusicFolder;
+import org.airsonic.player.service.search.IndexType.FieldNames;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.NumericField;
+import org.springframework.stereotype.Component;
+
+/**
+ * A factory that generates the documents to be stored in the index.
+ */
+@Component
+public class DocumentFactory {
+
+  /**
+   * Normalize the genre string.
+   * @param genre genre string
+   * @return genre string normalized
+   * @deprecated should be resolved with tokenizer or filter
+   */
+  @Deprecated
+  private String normalizeGenre(String genre) {
+    return genre.toLowerCase().replace(" ", "").replace("-", "");
+  }
+
+  /**
+   * Create a document.
+   * 
+   * @param mediaFile target of document
+   * @return document
+   * @since legacy
+   */
+  public Document createAlbumDocument(MediaFile mediaFile) {
+    Document doc = new Document();
+    doc.add(new NumericField(FieldNames.ID, Field.Store.YES, false).setIntValue(mediaFile.getId()));
+
+    if (mediaFile.getArtist() != null) {
+      doc.add(new Field(FieldNames.ARTIST, mediaFile.getArtist(), Field.Store.YES,
+          Field.Index.ANALYZED));
+    }
+    if (mediaFile.getAlbumName() != null) {
+      doc.add(new Field(FieldNames.ALBUM, mediaFile.getAlbumName(), Field.Store.YES,
+          Field.Index.ANALYZED));
+    }
+    if (mediaFile.getFolder() != null) {
+      doc.add(new Field(FieldNames.FOLDER, mediaFile.getFolder(), Field.Store.NO,
+          Field.Index.NOT_ANALYZED_NO_NORMS));
+    }
+    return doc;
+  }
+
+  /**
+   * Create a document.
+   * 
+   * @param mediaFile
+   *        target of document
+   * @return document
+   * @since legacy
+   */
+  public Document createArtistDocument(MediaFile mediaFile) {
+    Document doc = new Document();
+    doc.add(new NumericField(FieldNames.ID, Field.Store.YES, false).setIntValue(mediaFile.getId()));
+
+    if (mediaFile.getArtist() != null) {
+      doc.add(new Field(FieldNames.ARTIST, mediaFile.getArtist(), Field.Store.YES,
+          Field.Index.ANALYZED));
+    }
+    if (mediaFile.getFolder() != null) {
+      doc.add(new Field(FieldNames.FOLDER, mediaFile.getFolder(), Field.Store.NO,
+          Field.Index.NOT_ANALYZED_NO_NORMS));
+    }
+    return doc;
+  }
+
+  /**
+   * Create a document.
+   * 
+   * @param album
+   *        target of document
+   * @return document
+   * @since legacy
+   */
+  public Document createDocument(Album album) {

```

Comment:
Can't this one be made generic, an merged with `public Document createAlbumDocument(MediaFile mediaFile)` ?

Reply:
createAlbumDocument (MediaFile mediaFile) is an Album of File structure.
createDocument (Album album) is for Id3.
(Maybe it is hard to understand the name ...?)

Reply:
ok